src/main/java/com/onevizion/scmdb/DdlGenerator.java

```diff
@@ -451,19 +452,32 @@ public void generateDllsForAllDbObjects() {
     private String sortConstraintsInTableDdl(String sourceDdlScript) {
         Matcher constraintsBlockMatcher = CONSTRAINTS_BLOCK_PATTERN.matcher(sourceDdlScript);
         Map<String, String> constraintByName = new TreeMap<>(CONSTRAINT_COMPARATOR);
+        String primaryKey = null;
         if (constraintsBlockMatcher.find()) {
             String block = constraintsBlockMatcher.group(1);
             String[] constraints = block.split(",\\s*\\n");
             sourceDdlScript = sourceDdlScript.replace(block, "@");
-            for (String s : constraints) {
-                Matcher constraintNameMatcher = CONSTRAINT_NAME_PATTERN.matcher(s);
+            for (String constraintLine : constraints) {
+                if (primaryKey == null) {
```

Comment:
extract to the method 
<img width="990" alt="image" src="https://user-images.githubusercontent.com/3282060/230836113-d0344fb9-4894-4858-b6c1-282010138225.png">

Reply:
I divided the method into two logical parts and put the code into separate methods. 
Corrected `regx` for the name constants so that the primary key is not duplicated - due to moved of the primary key search  in a separate method.